### ISSUE TITLE
30 percentage of imagenet data

### ISSUE THREAD

**Issue by deeplearningmachine** [tensorpack/tensorpack#994]
The imagenet ResNet is too big for my GPU. How can I train it with only 30% of the original training data?

**Comment by ppwwyyxx** [comment]
Please read the tutorials about data.

**Comment by deeplearningmachine** [comment]
I did use some codes like below, but it seemed to use too much memory to load the data. I wonder if there is a way using less memory?

---
def get_data(train_or_test, fake=False):
    if fake:
        return FakeData([[64, INPUT_SHAPE, INPUT_SHAPE, 3], [64]], 1000, random=False, dtype='uint8')
    isTrain = train_or_test == 'train'
    datadir = args.data
    ds = dataset.ILSVRC12(datadir, train_or_test,
                          shuffle=True if isTrain else False, dir_structure='original')

    ds = FixedSizeData(ds,100000)

---

**Comment by ppwwyyxx** [comment]
By memory do you mean CPU memory?
You can use fewer prefetching processes in that case.

**Comment by deeplearningmachine** [comment]
No, it seemed it used too much GPU memory for that.

**Comment by ppwwyyxx** [comment]
DataFlow has nothing to do with GPU memory.

**Comment by deeplearningmachine** [comment]
I see. so, it may be caused by the image size? 
when I crop the image to smaller size, say 200 instead of 224 as your default value, the whole net can fit into my GPU. But with smaller image size, all the Resnet parameters such as learning rate may need to be retuned.

**Comment by ppwwyyxx** [comment]
You can use small images, small networks, small batch size, etc...

**Comment by ppwwyyxx** [comment]
I do not.

**Comment by ppwwyyxx** [comment]
As said in the [issue template](https://github.com/tensorpack/tensorpack/blob/master/.github/ISSUE_TEMPLATE.md), we do not answer __machine learning__ questions.

**Comment by deeplearningmachine** [comment]
Sorry; just tried to reproduce your benchmark accuracy with my limited GPU power. 

I will look into how you synchronize the results of the 8 GPUs and how you distribute the 256 samples in the batch to the 8 GPUs in your implementation. Thanks!
